pkg/cli/whoami/whoami.go

```diff
@@ -153,14 +164,34 @@ func (o *WhoAmIOptions) Run() error {
 		}
 		fmt.Fprintf(o.Out, "%s\n", consoleUrl)
 		return nil
-	}
+	case o.Output == "yaml":
+		u, err := o.UserInterface.Users().Get(context.TODO(), "~", metav1.GetOptions{})
+		u.SetGroupVersionKind(userv1.GroupVersion.WithKind("User"))
+		if err != nil {
+			return err
+		}
 
-	var err error
-	o.UserInterface, err = userv1typedclient.NewForConfig(o.ClientConfig)
-	if err != nil {
-		return err
+		y, err := yaml.Marshal(u)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(o.Out, "%s\n", string(y))
+		return nil
+	case o.Output == "json":
```

Comment:
We can use built-in printing flags to print yaml and json files. Example, https://github.com/kubernetes/kubectl/blob/5ff92a69e3ecb06912d5f468186d458bf610244e/pkg/cmd/auth/whoami.go#L64